Feat: #152 출차 v2 구현

dev-dykim opened the pull request:
## 📕 출차 v2 구현

## 📗 작업 내용

### PR 타입
- [x] 기능 추가
- [ ] 기능 삭제
- [ ] 버그 수정
- [ ] 코드 리팩토링

### 반영 브랜치
feat/#152-v2-exit -> version2

### 변경 사항
- 즉시예약, 사전예약 & 예약종료 전 출차하는 경우, 사전예약 & 예약종료 후 출차하는 경우 나누어 요금 계산 구현
- 출차시, 기존 시간별 예약현황에서 출차 후 시간대의 주차가능대수를 한 자리씩 늘리도록 구현

### 테스트 결과
Postman 테스트 결과 이상 없습니다.

dev-dykim committed:
Feat: #152 출차 v2 구현

- 즉시예약, 사전예약 & 예약종료 전 출차하는 경우, 사전예약 & 예약종료 후 출차하는 경우 나누어 요금 계산 구현
- 출차시, 기존 시간별 예약현황에서 출차 후 시간대의 주차가능대수를 한 자리씩 늘리도록 구현

Gem-o-b commented on src/main/java/com/sparta/parknav/management/service/MgtService.java:
```diff
@@ -141,26 +141,35 @@ public CarOutResponseDto exit(CarNumRequestDto requestDto, Admin user) {
+        // SCENARIO EXIT 5
+        int endHour = (bookingInfo.getEndTime().getMinute() == 0 && bookingInfo.getEndTime().getSecond() == 0) ? bookingInfo.getEndTime().getHour() - 1 : bookingInfo.getEndTime().getHour();
+        List<ParkBookingByHour> hourList = parkBookingByHourRepository
+                .findByParkInfoIdAndDateAndTimeBetween(parkOperInfo.getParkInfo().getId(), now.toLocalDate(), now.getHour(), endHour);
```
혹시 이 코드에서 1일에 들어온 차량이 3일간 예약을했는데 1시간만에 출차한경우도
2일,3일차 리스트를 정상적으로 가져올까요?

dev-dykim replied:
2일,3일차도 가져오도록 수정했습니다!

dev-dykim committed:
Feat: #152 출차시 요금계산 수정

- 스케쥴링에 의해 예약종료시간 이후에 출차하는 경우는 없으므로, 즉시예약, 사전예약 두 경우로 나누어 요금 계산

osy9536 reviewed (approved):
확인했습니다 👍

dev-dykim committed:
Fix: #152 시간별 예약가능대수 저장 로직 수정

- 하루를 넘겨 예약할 경우 시간별 예약가능대수가 제대로 저장되지 않아 모든 시간대 저장되도록 수정

dev-dykim committed:
Feat: #152 출차시 시간별 예약가능대수 업데이트 로직 구현

- 출차시간과 예약종료시간까지 모든 시간대의 예약가능대수를 업데이트하도록 구현
- 출차시간 ~ 예약종료시간이 당일인 경우, 하루이상 차이 나는 경우를 나누어 쿼리가 실행되도록 QueryDSL 사용

dev-dykim committed:
Chore: #152 불필요한 코드 삭제

Gem-o-b reviewed (approved):
수고하셨습니다👍
